src/components/shareable-url-modal/shareable-url-modal.js

```diff
@@ -125,83 +128,157 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
 
   const handleModalClose = () => {
     onToggleModal(false);
-    setDeploymentState('default');
+    if (deploymentState !== 'incompatible') {
+      setDeploymentState('default');
+    }
     setResponseError(null);
     setIsLoading(false);
     setResponseUrl(null);
     setIsLinkSettingsClick(false);
     setInputValues({});
     setIsFormDirty({
       bucket_name: false,
-      region: false,
+      platform: false,
+      endpoint: false,
     }); /* eslint-disable camelcase */
   };
 
+  /**
+   * Returns the modal heading based on the given type and deployment state.
+   * @param {string} type - The type of the modal heading.
+   * @returns {string|null} The modal heading text or null if deployment state is 'default'.
+   */
+  const getModalHeading = (type) => {
+    if (deploymentState === 'default') {
+      return null;
+    }
+
+    if (type === 'title') {
+      return deploymentState === 'success'
+        ? 'Kedro-Viz Published and Hosted'
+        : 'Publish and Share Kedro-Viz';
+    }
+
+    return modalMessages(deploymentState, compatibilityData.package_version);
+  };
+
+  const handleResponseUrl = () => {
+    // If the URL does not start with http:// or https://, append http:// to avoid relative path issue
```

Comment:
appending **http** will not work for Azure. By default AzureBlobStorage serves content on **https**
`Secure transfer required is enabled by default`

![image](https://github.com/kedro-org/kedro-viz/assets/87735534/e72ee386-7ad9-4c74-8163-8f9fbe263d5c)